libsolidity/interface/CompilerStack.cpp

```diff
@@ -571,26 +595,26 @@ string CompilerStack::absolutePath(string const& _path, string const& _reference
 }
 
 void CompilerStack::compileContract(
-	bool _optimize,
-	unsigned _runs,
 	ContractDefinition const& _contract,
 	map<ContractDefinition const*, eth::Assembly const*>& _compiledContracts
 )
 {
 	if (_compiledContracts.count(&_contract) || !_contract.annotation().isFullyImplemented)
 		return;
 	for (auto const* dependency: _contract.annotation().contractDependencies)
-		compileContract(_optimize, _runs, *dependency, _compiledContracts);
+		compileContract(*dependency, _compiledContracts);
 
-	shared_ptr<Compiler> compiler = make_shared<Compiler>(_optimize, _runs);
-	compiler->compileContract(_contract, _compiledContracts);
+	shared_ptr<Compiler> compiler = make_shared<Compiler>(m_optimize, m_optimizeRuns);
 	Contract& compiledContract = m_contracts.at(_contract.name());
+	string onChainMetadata = createOnChainMetadata(compiledContract);
+	compiler->compileContract(_contract, _compiledContracts, dev::swarmHash(onChainMetadata));
```

Comment:
If I acquire the metadata through a different channel (such as a registry, or a block explorer), I can still validate it's hash without the need of understanding a specific version of Swarm's hashing algorithm (which is subject to change, whereas `keccak256` is not).